src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -149,6 +149,20 @@ internal partial class Binder
                     unconvertedSource.HasErrors);
             }
 
+            if (conversion.Kind == ConversionKind.InterpolatedStringHandler)
+            {
+                var unconvertedSource = (BoundUnconvertedInterpolatedString)source;
+                return new BoundConversion(
+                    syntax,
+                    BindUnconvertedInterpolatedStringToHandlerType(unconvertedSource, (NamedTypeSymbol)destination, diagnostics, isHandlerConversion: true),
+                    conversion,
+                    @checked: false,
```

Comment:
Does this conversion cover the whole expression? If I have arithmetic in the interpolation holes for example?

Reply:
Any expression is valid in the interpolation holes, we don't treat anything special.

Reply:
Oh, I see what you're saying. It doesn't apply to anything inside interpolations, but we should probably use the value of `CheckOverflowAtRuntime` anyway.